### PR TITLE
add role auditor

### PR DESCRIPTION
**Pull request by QvodSoldier**
#### What type of this PR

Add one of the following kinds:
/kind feature

#### What this PR does / why we need it:
add a auditor role

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://terminus-org.app.terminus.io/erda/dop/projects/387/issues/all?id=208355&issueFilter__urlQuery=eyJpdGVyYXRpb25JRHMiOls1MDZdLCJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMDQzNiJdfQ%3D%3D&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=506&type=TASK)

#### Specified Reviewers:

/assign @Effet

### CODE REVIEW

File: modules/core-services/endpoints/member.go [hunk 1]
```diff
@@ -148,7 +148,8 @@ func (e *Endpoints) ListMemberRoles(ctx context.Context, r *http.Request, vars m
 			}
 		}
 	}
-	if orgID == 0 {
+
+	if orgID == 0 && scopeType != apistructs.SysScope {
```
Comment (Effet) [inline modules/core-services/endpoints/member.go:152]: different condition should return different error

File: modules/core-services/services/member/member.go [hunk 2]
```diff
@@ -602,7 +602,18 @@ func (m *Member) CheckPermission(userID string, scopeType apistructs.ScopeType,
...
+	if err != nil {
+		return err
+	}
+
 	switch scopeType {
+	case apistructs.SysScope:
+		// 系统管理员
+		if admin {
+			return nil
+		}
```
Comment (Effet) [inline modules/core-services/services/member/member.go:616]: why not check admin user's permission as same as other user?

I think you just query `m.db.GetMemberByScopeAndUserID(userID, scopeType, scopeID)` also work for admin?

thus `m.db.IsSysAdmin` may useless?

File: modules/dop/endpoints/member.go [hunk 3]
```diff
@@ -60,7 +60,8 @@ func (e *Endpoints) ListMemberRoles(ctx context.Context, r *http.Request, vars m
 			}
 		}
 	}
-	if orgID == 0 {
+
+	if orgID == 0 && scopeType != apistructs.SysScope {
```
Comment (Effet) [inline modules/dop/endpoints/member.go:64]: different condition should return different error

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/core-services/endpoints/member.go (hunk 1 above)
Comment (QvodSoldier) [reply modules/core-services/endpoints/member.go:152]: I think it's the same condition that means the orgID can't be equal to 0, only if the scopeType is 'sys'.

File: modules/dop/endpoints/member.go (hunk 3 above)
Comment (QvodSoldier) [reply modules/dop/endpoints/member.go:64]: I think it's the same condition that means the orgID can't be equal to 0, only if the scopeType is 'sys'.

File: modules/core-services/services/member/member.go (hunk 2 above)
Comment (QvodSoldier) [reply modules/core-services/services/member/member.go:616]: `m.db.GetMemberByScopeAndUserID(userID, scopeType, scopeID)` will return all the members in this scope, and if you use this function, you need to do another traversal outside.  I think this old function is also quite good.

**Comment by sfwn** [comment]
add unit test to match coverage requirement.

**Review by Effet — APPROVED** [review]

**Pull request merged**